crates/biome_css_analyze/src/lint/nursery/no_duplicate_selectors.rs

```diff
@@ -0,0 +1,327 @@
+    ///
+    /// ## Options
+    ///
+    /// If true, disallow duplicate selectors within selector lists.
+    ///
+    /// ```json
+    /// {
+    ///     "noDuplicateSelectors": {
+    ///         "options": {
+    ///           "disallowInList": true
+    ///         }
+    ///     }
+    /// }
+    /// ```
+    ///
+    pub NoDuplicateSelectors {
+        version: "next",
+        name: "noDuplicateSelectors",
+        recommended: true,
+        sources: &[RuleSource::Stylelint("no-duplicate-selectors")],
+    }
+}
+
+#[derive(Debug, Clone, Deserialize, Deserializable, Eq, PartialEq, Serialize)]
+#[cfg_attr(feature = "schemars", derive(schemars::JsonSchema))]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+pub struct NoDuplicateSelectorsOptions {
+    pub disallow_in_list: bool,
+}
+
+impl Default for NoDuplicateSelectorsOptions {
+    fn default() -> Self {
+        Self {
+            disallow_in_list: false,
+        }
+    }
+}
+
+#[derive(Debug, Eq)]
+struct ResolvedSelector {
+    selector_text: String,
+    selector_node: CssSyntaxNode,
+}
+
+impl PartialEq for ResolvedSelector {
+    fn eq(&self, other: &ResolvedSelector) -> bool {
+        self.selector_text == other.selector_text
+    }
+}
+impl Hash for ResolvedSelector {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        self.selector_text.hash(state);
+    }
+}
+impl Borrow<String> for ResolvedSelector {
+    fn borrow(&self) -> &String {
+        &self.selector_text
+    }
+}
+
+pub struct DuplicateSelector {
+    first: CssSyntaxNode,
+    duplicate: CssSyntaxNode,
+}
+
+impl Rule for NoDuplicateSelectors {
+    type Query = Ast<CssRoot>;
+    type State = DuplicateSelector;
+    type Signals = Vec<Self::State>;
+    type Options = NoDuplicateSelectorsOptions;
+
+    fn run(ctx: &RuleContext<Self>) -> Vec<Self::State> {
+        let node = ctx.query();
+        let options = ctx.options();
+
+        let mut resolved_list: HashSet<ResolvedSelector> = HashSet::new();
+        let mut output: Vec<DuplicateSelector> = vec![];
+
+        if options.disallow_in_list {
+            let selectors = node.rules().syntax().descendants().filter(|x| {
+                x.clone().cast::<AnyCssSelector>().is_some()
+                    || x.clone().cast::<AnyCssRelativeSelector>().is_some()
+            });
+
+            // selector_list unwrap should never fail due to the structure of the AST
+            for (selector, selector_list) in selectors
+                .map(|selector| (selector.clone(), selector.parent().unwrap()))
+                .filter(|(_, parent)| {
+                    // i.e not actually a list
+                    return !(parent.clone().cast::<CssComplexSelector>().is_some()
+                        || parent.clone().cast::<CssRelativeSelector>().is_some());
+                })
+            {
+                // this_rule unwrap should never fail due to the structure of the AST
+                let this_rule = selector_list.parent().unwrap();
```

Comment:
Unfortunately, this is never guaranteed, because another rule could create an invalid CST. 

I suggest the following:

```rs
let Some(this_rule) = selector_list.parent() else {
	continue;
}
```